src/Adapter/CustomerService/CommandHandler/ReplyToCustomerThreadHandler.php

```diff
@@ -0,0 +1,174 @@
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\CustomerService\CommandHandler;
+
+use Contact;
+use Context;
+use Customer;
+use CustomerMessage;
+use CustomerThread;
+use Language;
+use Mail;
+use PrestaShop\PrestaShop\Core\Domain\CustomerService\Command\ReplyToCustomerThreadCommand;
+use PrestaShop\PrestaShop\Core\Domain\CustomerService\CommandHandler\ReplyToCustomerThreadHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\CustomerService\Exception\CustomerServiceException;
+use PrestaShop\PrestaShop\Core\Domain\CustomerService\ValueObject\CustomerThreadStatus;
+use ShopUrl;
+use Symfony\Component\Translation\TranslatorInterface;
+use Tools;
+use Validate;
+
+/**
+ * @internal
+ */
+final class ReplyToCustomerThreadHandler implements ReplyToCustomerThreadHandlerInterface
+{
+    /**
+     * @var Context
+     */
+    private $context;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @param Context $context
+     */
+    public function __construct(Context $context)
+    {
+        $this->context = $context;
+        $this->translator = $context->getTranslator();
+    }
+
+    /**
+     * @param ReplyToCustomerThreadCommand $command
+     */
+    public function handle(ReplyToCustomerThreadCommand $command)
+    {
+        $customerThread = new CustomerThread(
+            $command->getCustomerThreadId()->getValue()
+        );
+
+        ShopUrl::cacheMainDomainForShop((int) $customerThread->id_shop);
+
+        $this->createCustomerMessage(
+            $customerThread,
+            $command->getReplyMessage()
+        );
+
+        $replyWasSent = $this->sendReplyEmail($customerThread, $command);
+
+        if ($replyWasSent) {
+            $customerThread->status = CustomerThreadStatus::CLOSED;
+            $customerThread->update();
+        }
+    }
+
+    /**
+     * @param CustomerThread $customerThread
+     * @param string $replyMessage
+     *
+     * @return CustomerMessage
+     */
+    private function createCustomerMessage(CustomerThread $customerThread, $replyMessage)
```

Comment:
Almost identical to https://github.com/PrestaShop/PrestaShop/pull/14288/files#diff-17c9f8d5fbb4a31653445434c3d9493bR308

Reply:
Yup, but that's the catch - _almost_. I feel that keeping it separate is more logical thus it follows SRP instead of abstracting it and introducing behavior toggling argument on method. Wdyt?

Reply:
Makes sense 👍